src/api/app/components/notification_component.html.haml

```diff
@@ -0,0 +1,40 @@
+.list-group-item.px-0.px-md-1.py-3
+  .row
+    .col-auto.pr-0
+      .custom-control.custom-checkbox
+        = check_box_tag('notification_ids[]', @notification.id, false,
+                        id: "notification_ids_#{@notification.id}", class: 'custom-control-input')
+        = label_tag("notification_ids_#{@notification.id}", '', class: 'custom-control-label')
+    .col-10
+      .row
+        .col
+          - if @notification.notifiable_type == 'BsRequest'
+            = image_tag('icons/request-icon.svg', height: 18, title: 'Request notification')
+            = render NotificationNotifiableLinkComponent.new(@notification)
+            %small.text-nowrap #{time_ago_in_words(@notification.created_at)} ago
+            %span.badge.ml-1{ class: "badge badge-#{helpers.request_badge_color(@notification.notifiable.state)}" }
+              = @notification.notifiable.state
+          - else
+            %i.fas.fa-comments{ title: 'Comment notification' }
+            = render NotificationNotifiableLinkComponent.new(@notification)
+            %small.text-nowrap #{time_ago_in_words(@notification.created_at)} ago
+        .col-auto.actions.ml-auto.align-self-end.align-self-md-start
+          :ruby
+            title, icon = @notification.unread? ? ['Mark as "Read"', 'fa-check'] : ['Mark as "Unread"', 'fa-undo']
```

Comment:
I think this can also be pushed up into the component so we only ask for the title and the icon in the view, and the component does the logic :thought_balloon:

Reply:
I added a `NotificationMarkButtonComponent` for this.